cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -160,6 +175,20 @@ public void startTx(Transaction tx) {
   }
 
   @Override
+  public boolean commitTx() throws Exception {
+    if (!super.commitTx()) {
+      return false;
+    }
+
+    // Publish a notification if there was new partition added.
+    RuntimeProgramContext runtimeProgramContext = getRuntimeProgramContext();
+    if (runtimeProgramContext != null && !partitionsAddedInSameTx.isEmpty()) {
```

Comment:
can you leverage `operationsInThisTx` instead of `partitionsAddedInSameTx`?
That way, we won't need to keep track of both collections?